src/Orckestra.Composer.Website/UI.Package/Typescript/Composer.MyAccount/UpdateAccount/UpdateAccountController.ts

```diff
@@ -8,70 +8,99 @@
 
 module Orckestra.Composer {
+    enum UpdateAccountStates {
+        Sucsess = 'sucsess',
+        Failed = 'failed'
+    }
 
     export class UpdateAccountController extends Orckestra.Composer.MyAccountController {
 
         protected customerService: ICustomerService = new CustomerService(new CustomerRepository());
         protected userService: UserMetadataService = new UserMetadataService(new MembershipRepository());
+        protected VueUpdateAccount: Vue;
 
         public initialize() {
 
             super.initialize();
-            this.registerSubscriptions();
+            let userData = this.context.viewModel;
+            const self = this;
+            const updateAccountFormId = "#updateAccountForm";
+
+            this.VueUpdateAccount = new Vue({
+                el: '#vueUpdateAccount',
+
+                data: {
+                    ...userData,
+                    UpdateAccountState: "",
+                    IsLoading: false
+                },
+                mounted() {
+                    self.initializeParsey(updateAccountFormId);
+                },
+                computed: {
+                    SaveChangesFailed() {
+                        return this.UpdateAccountState === UpdateAccountStates.Failed;
+                    },
+                    SaveChangesSucceeded() {
+                        return this.UpdateAccountState === UpdateAccountStates.Sucsess;
+                    }
+                },
+                methods: {
+                    enableSubmitButton() {
+                        this.UpdateAccountState = "";
+                        let parsleyInit = self.getParsleyInit(updateAccountFormId);
+                        $('#UpdateAccountSubmit').prop('disabled', !parsleyInit.isValid());
+                    },
+                    updateAccount() {
+                        let parsleyInit = self.getParsleyInit(updateAccountFormId);
+                        if (parsleyInit && !parsleyInit.validate()) { return; }
+
+                        var formData: any = {
+                            FirstName: this.FirstName,
+                            LastName: this.LastName,
+                            Email: this.Email,
+                            PreferredLanguage: this.Language
+                        };
+
+                        var returnUrlQueryString: string = 'ReturnUrl=';
+                        var returnUrl: string = '';
+
+                        if (window.location.href.indexOf(returnUrlQueryString) > -1) {
+                            returnUrl = urlHelper.getURLParameter(location.search, 'ReturnUrl');
+                        }
+
+                        this.IsLoading = true;
+                        self.customerService.updateAccount(formData, returnUrl)
+                            .then((result) => {
+                                self.userService.invalidateCache();
+                                self.onAccountUpdated(result);
```

Comment:
before   self.onAccountUpdated(result); was exucuted by subscription on event MyAccountEvents.AccountUpdated,
now you execute it , and then publish event
return logic as it was before